client/components/translator-invite/index.jsx

```diff
@@ -28,50 +29,63 @@ export class TranslatorInvite extends Component {
 	recordClick = () =>
 		this.props.recordTracksEvent( 'calypso_translator_invitation', {
 			language: this.props.localizedLanguageNames[ this.props.locale ].en,
+			location: this.props.path,
 		} );
 
 	renderNoticeLabelText() {
 		const { locale, localizedLanguageNames, translate } = this.props;
 
-		if ( localizedLanguageNames && localizedLanguageNames[ locale ] ) {
-			return (
-				<div className="translator-invite__content">
-					<Gridicon className="translator-invite__gridicon" icon="globe" size={ 18 } />
-					{ translate(
-						'Would you like to help us translate WordPress.com into {{a}}%(language)s{{/a}}?',
-						{
-							args: { language: localizedLanguageNames[ locale ].localized },
-							comment:
-								'The language variable can be any major spoken language that WordPress.com supports',
-							components: {
-								a: (
-									<a
-										href={ `https://translate.wordpress.com/projects/wpcom/${ locale }/default/` }
-										onClick={ this.recordClick }
-										target="_blank"
-										rel="noopener noreferrer"
-									/>
-								),
-							},
-						}
-					) }
-				</div>
-			);
+		if ( ! localizedLanguageNames || ! localizedLanguageNames[ locale ] ) {
+			return null;
 		}
 
-		return null;
+		const languageName = localizedLanguageNames[ locale ].localized;
+		const currentLanguage = languages.find( ( language ) => language.langSlug === locale );
+		const percentTranslated = currentLanguage?.calypsoPercentTranslated || 0;
+
+		// translators: '%(languageName)s is a localized language name, %(percentTranslated)d%% is a percentage number (0-100), followed by an escaped percent sign %%'
+		const noticeText = translate(
+			'%(languageName)s is only %(percentTranslated)d%% translated. Help translate WordPress into your language.',
+			{
+				args: { languageName, percentTranslated },
+			}
+		);
+
+		return <div className="translator-invite__content">{ noticeText }</div>;
 	}
 
 	render() {
-		const { locale } = this.props;
-
+		const { locale, translate } = this.props;
 		if ( config( 'i18n_default_locale_slug' ) === locale ) {
 			return null;
 		}
 
+		const currentLanguage = languages.find( ( language ) => language.langSlug === locale );
+		const isTranslatedIncompletely = currentLanguage?.isTranslatedIncompletely;
+
 		return (
 			<div className="translator-invite">
-				{ this.renderNoticeLabelText() }
+				{ isTranslatedIncompletely && (
```

Comment:
The component will render the empty `translator-invite` div and the `<QueryLanguageNames />` for languages with complete languages, which I suppose is not intended.

Reply:
Actually, this was done to preserve existing behaviour, and fix [this test](https://github.com/Automattic/wp-calypso/blob/8464223a66e00048511ce73b27f39e02093a1742/client/components/translator-invite/test/index.js#L37).

Reply:
I don't really see the point to render an empty DOM element in this case. Also, requesting the language names data that is not going to be used because the rendering condition is not fulfilled seems pointless too.